### PR TITLE
solution to antisnake

### PR DESCRIPTION
**Pull request by mykola-koval**
- [DEMO LINK](https://mykola-koval.github.io/layout_antisnake/)
- [TEST REPORT LINK](https://mykola-koval.github.io/layout_antisnake/report/html_report/)

### CODE REVIEW

**Review by AlexAzalor — APPROVED** [review]

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+  .main {
+    grid-template-columns: repeat(2, 1fr);
+  }
+}
+
+@media (min-width: 900px) {
+  .main {
+    grid-template-columns: repeat(3, 1fr);
+  }
 }
```
Comment (Elena-brv) [inline src/styles/main.scss:42]: you can nake this task with a 1 media-query, and use nesting for `@media`

File: src/styles/utils/_variables.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+$parts:
```
Comment (Elena-brv) [inline src/styles/utils/_variables.scss:1]:
```suggestion
$colors:
```

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+
+    @each $partnumber, $color in $parts {
+      &:nth-child(#{$partnumber}) {
+        background-color: $color;
+      }
+
+      &:nth-child(#{$partnumber})::before {
+        content: "#{$partnumber}";
+        color: white;
+      }
```
Comment (Elena-brv) [inline src/styles/main.scss:27]:
```suggestion
      &:nth-child(#{$partnumber}) {
        background-color: $color;
        
        &::before {
          content: "#{$partnumber}";
          color: white;
        }
```

**Review by Elena-brv — CHANGES_REQUESTED** [review]

File: src/styles/utils/_variables.scss (hunk 2 above)
Comment (mykola-koval) [reply src/styles/utils/_variables.scss:1]: it is not just colors, but also the part numbers combined

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+        }
+      }
+    }
+  }
+
+  @media (min-width: 900px) {
+    & {
+      grid-template-columns: repeat(3, 1fr);
+    }
+  }
```
Comment (Elena-brv) [inline src/styles/main.scss:35]:
```suggestion
  @media (min-width: 900px) {
    grid-template-columns: repeat(3, 1fr);
  }
```

**Review by Elena-brv — APPROVED** [review]

**Pull request closed without merging**